packages/react-core/src/components/CalendarMonth/CalendarMonth.tsx

```diff
@@ -217,6 +222,34 @@ export const CalendarMonth = ({
   const changeMonth = (newMonth: number, newYear?: number) =>
     new Date(newYear ?? focusedDate.getFullYear(), newMonth, 1);
 
+  const MIN_YEAR = 1900;
+  const MAX_YEAR = 2100;
+
+  const handleYearInputChange = (event: React.FormEvent<HTMLInputElement>, yearStr: string) => {
+    if (!/^\d{0,4}$/.test(yearStr)) {
+      return;
+    }
+
+    setYearInput(yearStr);
+
+    if (yearStr.length === 4) {
+      const yearNum = Number(yearStr);
+
+      if (yearNum >= MIN_YEAR && yearNum <= MAX_YEAR) {
+        const newDate = changeYear(yearNum);
+        setFocusedDate(newDate);
+        setHoveredDate(newDate);
+        setShouldFocus(false);
+
+        // We need to unfocus a date when changing year via up/down arrows in FireFox, otherwise focus doesn't go to the year input like expected
+        focusRef.current?.blur();
```

Comment:
Could've sworn in testing yesterday that this worked as originally intended. Noticing in looking again this morning that firing this blur when clicking the scroll buttons in FF has some new(?) odd behavior:

https://github.com/patternfly/patternfly-react/assets/70952936/101078cf-cf6f-4c56-a5e1-5965262f6fc1

Maybe I had just happened to test when the "next month" button received focus correctly.

TLDR is instead we could replace this blur call on the focusRef, and instead `focus()` the TextInput:

```suggestion
        // We need to manually focus the year input in FireFox when the scroll buttons are clicked, as FireFox doesn't place focus automatically
        (event.target as HTMLElement).focus();
```
Tested this a few more times in FF and Chrome and didn't notice any similar bug to the video above.